downloader.go

```diff
@@ -35,6 +35,7 @@ func (dl *Downloader) initiateAPI() error {
 func (dl *Downloader) DownloadSchedule() (playlist.Schedule, error) {
 	if err := dl.initiateAPI(); err != nil {
 		log.Printf("Could not connect to api. %s", err)
+		return playlist.Schedule{}, err
```

Comment:
Maybe don't worry about this right now, but a good general guide is to either log an error or return it but not both. Otherwise you risk the same error being logged at multiple levels which can be confusing.

Where possible I find it's useful to restrict logging output to the higher layers of the code.